docs/source/deployguide/ca-config.md

```diff
@@ -0,0 +1,423 @@
+#  Checklist for a production CA server
+
+As you prepare to build a production Fabric CA server, you need to consider configuration of the following fields in the fabric-ca-server-config.yaml file.
+When you [initialize](https://hyperledger-fabric-ca.readthedocs.io/en/release-1.4/users-guide.html#initializing-the-server) the CA server, this file is generated for you so that you can customize it before actually starting the server.
+
+This checklist covers key configuration parameters for setting up a production network. Of course you can always refer to the [fabric-ca-server-config.yaml
+file](https://hyperledger-fabric-ca.readthedocs.io/en/release-1.4/serverconfig.html) for additional parameters or more information. The list of parameters in this topic includes:
+
+- [ca](#ca)
+- [tls](#tls)
+- [cafiles](#cafiles)
+- [intermediate ca](#intermediate-ca)
+- [port](#port)
+- [user registry](#user-registry)
+- [registry database](#registry-database)
+- [LDAP](#ldap)
+- [affiliations](#affiliations)
+- [csr](#csr)
+- [signing](#signing)
+- [bccsp](#bccsp)
+- [cors](#cors)
+- [cfg](#cfg)
+- [operations](#operations)
+- [metrics](#metrics)
+
+## ca
+
+```
+ca:
+  # Name of this CA
+  name:
+  # Key file (is only used to import a private key into BCCSP)
+  keyfile:
+  # Certificate file (default: ca-cert.pem)
+  certfile:
+  # Chain file
+  chainfile:
+```
+
+Start by giving your CA a name. Often the name indicates the organization that this CA will serve. Or, if this is a TLS CA you might want to indicate that in the name. This `ca.name` is used when targeting this server for requests by the Fabric CA Client `--caname` parameter.
+
+## tls
+
+```
+tls:
+  # Enable TLS (default: false)
+  enabled: true
+  # TLS for the server's listening port
+  certfile:
+  keyfile:
+  clientauth:
+    type: noclientcert
+    certfiles:
+```
+
+Configure this section to enable TLS communications for the organization.
+
+- **`tls.enabled`**: For a secure production environment, TLS should be enabled for secure communications between nodes by setting `enabled: true` in the `tls` section of the config file. (Note that it is disabled by default which may be acceptable for a test network but for production it needs to be enabled.) This setting will configure `server-side TLS`, meaning that TLS will guarantee the identity of the _server_ to the client and provides a two-way encrypted channel between them.
+
+- **`tls.certfile`**: Every node in the organization needs to enroll with the TLS CA for the organization before it can transact securely with other nodes in the organization. Therefore, before you can deploy an enrollment CA or an intermediate CA, you must first enroll the bootstrap identity with the TLS CA to get the TLS signing certificate. Provide the name and location of the generated TLS signing certificate. For example: `/msp/signcerts/cert.pem`. If this is a root TLS CA, this field can be blank.
```

Comment:
Every CA needs to register and enroll with its TLS CA before it can transact securely with other nodes in the organization. Therefore, before you can deploy an organization CA or an intermediate CA, you must first deploy the TLS CA and then register and enroll the organization CA bootstrap identity with the TLS CA to generate the organziation CA's TLS signed certificate. When you use the Fabric CA client to generate the certificates, the TLS signed cert is generated in the `signedcerts` folder of the specified msp directory of the `FABRIC_CA_CLIENT_HOME` folder. For example: `/msp/signcerts/cert.pem`.